common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -332,31 +331,35 @@ public static long parseSecondNano(String secondNano) throws IllegalArgumentExce
   }
 
   public final int months;
+  public final int days;
   public final long microseconds;
 
   public long milliseconds() {
```

Comment:
The method will return different values because you exclude days from microseconds